poglink/cogs/rates.py

```diff
@@ -170,14 +173,16 @@ async def on_ready(self):
                     logger.error(f"Failed to write rates to {output_path}: {e}")
                 except Exception as e:
                     logger.error(e)
-
-                if rates_diff.items:
-                    # generate and send embed
-                    logger.info(f"Rates at {url} changed - sending embed")
-                    embed_description = rates_diff.to_embed()
-                    await self.send_embed(embed_description, url)
-                else:
-                    logger.debug(f"No change in rates at {url}.")
+                try:
+                    if rates_diff.items:
+                        # generate and send embed
+                        logger.info(f"Rates at {url} changed - sending embed")
+                        embed_description = rates_diff.to_embed()
+                        await self.send_embed(embed_description, url)
+                    else:
+                        logger.debug(f"No change in rates at {url}.")
+                except Exception as e:
+                    raise RatesReadError(e) from e 
```

Comment:
I'm not following what this is supposed to be catching here, but I'm suspicious of calling it a `RatesReadError`, as this block should do no reading of any sort. Though this is my bad for not writing any docstrings to help guide how I envisioned any of the new custom exceptions to be interpreted.

Reply:
Exception copied from [previous use](https://github.com/FM-17/poglink/pull/40/commits/03fa7a2fa66b7db6c5f5a5798a5789f2ebf3cc91#r772044058) due to matching condition (`if rates_diff.items:`)

Reply:
Removed `RatesReadError` in favor of #44